vars/spinInfra.groovy

```diff
@@ -7,127 +7,156 @@ import uk.gov.hmcts.contino.MetricsPublisher
 import uk.gov.hmcts.pipeline.AKSSubscriptions
 import uk.gov.hmcts.contino.RepositoryUrl
 
-def call(productName, environment, tfPlanOnly, subscription) {
-  call(productName, null, environment, tfPlanOnly, subscription)
-}
-
-def call(product, component, environment, tfPlanOnly, subscription) {
-  call(product, component, environment, tfPlanOnly, subscription, "")
-}
-
-def call(product, component, environment, tfPlanOnly, subscription, deploymentTarget) {
-  def branch = new ProjectBranch(env.BRANCH_NAME)
+Closure infraArgs(Map args = [:]) {
+  def config = [
+          product                    : params.product,
```

Comment:
don't specify parameters that you aren't overriding they will be added by the `<< args` operation at the end of your array definition